submissions/OlStaniev/OOP exercise/index.js

```diff
@@ -0,0 +1,43 @@
+class Inhabitant {
+   constructor(specie, name, gender, legs, saying) {
+      this.specie = specie
+      this.name = name
+      this.gender = gender
+      this.legs = legs
+      this.saying = saying
+   }
+   prepareToPrint() {
+      return [this.specie, this.name, this.gender, this.legs, this.saying].join('; ')
+   }
+}
+
+class Person extends Inhabitant {
+   constructor(name, gender, legs, saying, hands, friends) {
+      super(name, gender, legs, saying)
+      this.hands = hands
+      this.friends = friends || []
+   }
+   prepareToPrint() {
+      return 'human; ' + super.prepareToPrint() + [this.hands, this.friends.join(', ')].join('; ')
```

Comment:
We won't need to hardcode species here. We have a base class to handle it. We just need to let it know what species we deal with.